### PR TITLE
Add a SidePanel control and support for displaying the layer switcher in a pane thereof

### PR DESCRIPTION
**Pull request by symbioquine**
**Why?** farmOS-map needs a place to organize extra controls and settings without continuing to clutter the map edges.

Resolves: https://github.com/farmOS/farmOS-map/issues/14

### CODE REVIEW

**Comment by symbioquine** [comment]
See whether it makes sense to have those dependencies as non-dev dependencies given that the way farmOS pulls the library from Asset Packagist.

**Comment by symbioquine** [comment]
Also check on the styling (alignment) of the layer switcher in the side panel.

**Comment by symbioquine** [comment]
> See whether it makes sense to [still] have those dependencies as non-dev dependencies given that the way farmOS pulls the library from Asset Packagist.

I've confirmed that when farmOS includes farmOS-map via Composer from Asset Packagist it also includes farmOS-map's transitive non-dev dependencies.

image

Unfortunately, I'm not sure it's practical to change them to be only dev dependencies since doing so would mean that any other NPM packages (e.g. FieldKit) trying to build against the farmOS-map source would need to duplicate those dependency declarations.

Curious to hear your thoughts on this @mstenta.

> Also check on the styling (alignment) of the layer switcher in the side panel.

For reference, the issue we're talking about is the left-justification of the layer tree where base layers appear less indented than the other layers;

image

As I pointed out on the call, this is actually an existing issue with the layer switcher;

image

And also appears in walkermatt's [LayerSwitcher sidebar] demo;

image

*Though there it looks a little less weird because the item with the carrot appears first.*

I'm not opposed to fixing this styling problem, but I do think it should be a separate follow-up to this PR. Any objections to that strategy @mstenta?

**Comment by mstenta** [comment]
> I've confirmed that when farmOS includes farmOS-map via Composer from Asset Packagist it also includes farmOS-map's transitive non-dev dependencies.

Since this is really an issue with farmOS itself and the way we're including farmOS-map via Asset Packagist, I don't think it needs to block this.

@paul121 had found a Composer package that we might be able to consider as an alternative to Asset Packagist, so we can think about that as the "solution" to this concern I think. I'll try to dig up that conversation and open an issue.

> I'm not opposed to fixing this styling problem, but I do think it should be a separate follow-up to this PR. Any objections to that strategy @mstenta?

No objections! Thanks for exploring it @symbioquine.

File: src/behavior/layerSwitcherInSidePanel.css [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.ol-side-panel-pane .layer-switcher {
+  left: 0px;
+}
+
+.ol-side-panel-pane .layer-switcher > ul {
+  margin-top: 0px;
+}
```
Comment (mstenta) [inline src/behavior/layerSwitcherInSidePanel.css:7]: Needs a newline.

File: src/behavior/sidePanel.css [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+@media (min-width : 768px) {
+  .map-with-ol-side-panel .ol-geocoder {
+    margin-left: 0;
+  }
+
+  .map-with-ol-side-panel.ol-side-panel-collapsed .ol-geocoder {
+    margin-left: 0px;
+  }
+}
```
Comment (mstenta) [inline src/behavior/sidePanel.css:27]: Needs a newline.

**Comment by mstenta** [comment]
Added two small newline nitpicks - but other than that I think this looks good!

**Comment by mstenta** [comment]
> @paul121 had found a Composer package that we might be able to consider as an alternative to Asset Packagist, so we can think about that as the "solution" to this concern I think. I'll try to dig up that conversation and open an issue.

FYI here is that conversation in the #farmOS chat logs: https://irc.farmos.org/bot/log/farmOS/2021-06-17#T65062

**Comment by symbioquine** [comment]
I rebased and force-pushed an updated version with some CSS cleanup including the requested newlines;

```patch
diff --git a/src/behavior/layerSwitcherInSidePanel.css b/src/behavior/layerSwitcherInSidePanel.css
index cb70be1..7ddb7f8 100644
--- a/src/behavior/layerSwitcherInSidePanel.css
+++ b/src/behavior/layerSwitcherInSidePanel.css
@@ -1,7 +1,7 @@
 .ol-side-panel-pane .layer-switcher {
-  left: 0px;
+  left: 0;
 }
 
 .ol-side-panel-pane .layer-switcher > ul {
-  margin-top: 0px;
-}
\ No newline at end of file
+  margin-top: 0;
+}
diff --git a/src/behavior/sidePanel.css b/src/behavior/sidePanel.css
index ef8dac3..db95e6c 100644
--- a/src/behavior/sidePanel.css
+++ b/src/behavior/sidePanel.css
@@ -4,8 +4,8 @@
 }
 
 .ol-side-panel .ol-side-panel-tabs button.active {
-  background-color: #ffcc33;
-  background-color: rgba(255,204,51,0.7);
+  background-color: #fc3;
+  background-color: rgba(255, 204, 51, 0.7);
 }
 
 .map-with-ol-side-panel .ol-geocoder {
@@ -13,15 +13,15 @@
 }
 
 .map-with-ol-side-panel.ol-side-panel-collapsed .ol-geocoder {
-  margin-left: 0px;
+  margin-left: 0;
 }
 
-@media (min-width : 768px) {
+@media (min-width: 768px) {
   .map-with-ol-side-panel .ol-geocoder {
     margin-left: 0;
   }
 
   .map-with-ol-side-panel.ol-side-panel-collapsed .ol-geocoder {
-    margin-left: 0px;
+    margin-left: 0;
   }
-}
\ No newline at end of file
+}
```

**Comment by symbioquine** [comment]
I've also opened #125 to enforce some basic CSS style rules so things like those newline comments don't need to slow PRs in the future.

**Comment by mstenta** [comment]
Awesome! Looks good to me! I think this is ready to merge @symbioquine

**Pull request merged**